src/pip/_internal/index.py

```diff
@@ -1143,7 +1143,7 @@ def _find_name_version_sep(egg_info, canonical_name):
     raise ValueError("{} does not match {}".format(egg_info, canonical_name))
 
 
-def _egg_info_matches(egg_info, canonical_name):
+def _extract_version(egg_info, canonical_name):
```

Comment:
nit: `_extract_version_from_egg_info`

Reply:
I’m not sure from_egg_info is a good name because this code path is used even when no “egg fragment” is found. Is the file base name still considered “egg info” in that case? I think from_base_name might be a safer bet if a descriptor is needed.

Reply:
(sorry for terseness)

I'm nitpicking; happy to go with whatever you prefer here. :)